packages/message-manager/src/AbstractMessageManager.ts

```diff
@@ -33,6 +34,7 @@ export interface AbstractMessage {
   type: string;
   rawSig?: string;
   securityProviderResponse?: Record<string, Json>;
+  metadata?: Record<string, string>;
```

Comment:
Would it be beneficial to be less restrictive with the types here?

Perhaps just `Json` so it's compatible with `BaseControllerV2` state, but also supports very simple use cases were the client puts a single value in here?

If the latter is too far, maybe just `Record<string, Json>` so we're not confined to string values?

Reply:
Ah! I was unaware of `Json` - I've changed the type here